### PR TITLE
Introducing some limits to avoid high memory consumption in corner cases

### PR DESCRIPTION
**Pull request by ivanprado**
Text and css generated features are unlimited, and they can be very big in some cases with a lot of text or nodes. Also, some pages have a very high number of links. This PR imposes some limits over them to avoid excessive memory consumption in such a case. 

Example of pages with such memory consumption (especially the first one): 
- https://www.lego.com/de-de/kids 
- https://www.cotswoldoutdoor.com/c/sale/shop-all.html
- https://www.nobodyschild.com/clothing.html

Based on the stats below and some runs, these are the default limits:

* max_text_features=80
* max_css_features=120
* max_links=2500

```
Stats for number of links (run over ~3000 pages):

             links
count  2986.000000
mean    331.445747
std     363.960655
min       0.000000
50%     237.500000
75%     403.000000
90%     664.000000
95%     899.750000
98%    1277.600000
99%    1602.600000
max    6592.000000

Stats for number of text and css features:

                text        classes
count  989697.000000  989697.000000
mean       35.574637      43.741996
std       206.472999      78.599356
min         0.000000       0.000000
50%        26.000000      22.000000
75%        42.000000      58.000000
90%        64.000000     108.000000
95%        84.000000     150.000000
98%       122.000000     226.000000
99%       158.000000     310.000000
max     65507.000000    5863.000000
```

Besides, this PR contains some changes already on Memex upstream: scikity 0.18 and eli5. 

Todo:

- [ ] Final run

### CODE REVIEW

**Comment by kmike** [comment]
~Hey @ivanprado! SH's fork is pretty old, there are some changes https://github.com/TeamHG-Memex/autopager - could you please send a PR to TeamGH-Memex repo instead?~ Sorry, you already sent it :)

**Comment by kmike** [comment]
btw, feel free to drop Python 2.x support.

**Comment by ivanprado** [comment]
@kmike any tips about how is the release process. Just execute `_release.sh` once changes are merged in master? I guess some rights are required.

**Comment by ivanprado** [comment]
Managing this change directly in memex repo
